.github/workflows/cmake-build.yml

```diff
@@ -0,0 +1,48 @@
+name: CMake Build
+
+on: [ push, pull_request ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions: { }
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'windows-latest', 'ubuntu-latest' ] #, 'macos-latest' ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Get cmake and ninja
+      uses: lukka/get-cmake@latest
```

Comment:
Why use latest rather than a specific version? Not great for repeatable builds.

Reply:
Are you're referring to the CMake/Ninja versions or the action's version?  Either way, you are right and it's not just about repeatability, but potentially about security too.  For a project like this, all the actions should probably use the [commit id](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) for referencing the version.

I'll take a look after getting all the presets to work.